Objects/bytearrayobject.c

```diff
@@ -1695,6 +1695,13 @@ bytearray_extend(PyByteArrayObject *self, PyObject *iterable_of_ints)
     int value;
     char *buf;
 
+    if (PyUnicode_Check(iterable_of_ints)) {
```

Comment:
This check is in the fast path and `PyUnicode_Check` checks for subtypes. Does it slow down the normal case?

Reply:
Why don't you put the check in the error path instead? I.e. replace the error popping up from `_getbytevalue`.

Reply:
done